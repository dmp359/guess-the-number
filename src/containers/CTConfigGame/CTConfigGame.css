.config {
  padding-top: 20px;
}

.config-label {
  font-weight: bold;
  padding: 20px 0;
}
