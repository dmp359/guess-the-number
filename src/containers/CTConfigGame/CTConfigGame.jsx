import React, { Component } from 'react';
import LabelBox from 'components/LabelBox/LabelBox.jsx';
import PropTypes from 'prop-types';

import './CTConfigGame.css';

/*
* Container for configuring gameplay
*/
class CTConfigGame extends Component {
  constructor(props) {
    super(props);
    this.state = { userLower: '', userUpper: '' };

    this.onChangeLower = this.onChangeLower.bind(this);
    this.onChangeUpper = this.onChangeUpper.bind(this);
    this.onReset = this.onReset.bind(this);
  }

  // On change of input boxes
  onChangeLower(e) {
    this.setState({ userLower: parseInt(e.target.value, 10) });
  };

  onChangeUpper(e) {
    this.setState({ userUpper: parseInt(e.target.value, 10) });
  };

  // On press of reset button
  onReset() {
    const { userLower, userUpper } = this.state;
    const lower = userLower || this.props.lower;
    const upper = userUpper || this.props.upper;
    this.props.onReset(lower, upper);
  }

  render() {
    const { lower, upper } = this.props;
    const { userLower, userUpper } = this.state;
    return (
      <div className='config'>
        <div className='config-label'>Game Config</div>
        <LabelBox
          label="Lower bound: "
          onChange={this.onChangeLower}
          value={userLower || lower} 
        />
        <LabelBox
          label="Upper bound: "
          onChange={this.onChangeUpper}
          value={userUpper || upper} 
        />
        <button onClick={this.onReset}>Reset</button>
      </div>
    );
  }
}

CTConfigGame.propTypes = {
  lower: PropTypes.number.isRequired,
  onReset: PropTypes.func.isRequired,
  upper: PropTypes.number.isRequired,
}

export default CTConfigGame;
