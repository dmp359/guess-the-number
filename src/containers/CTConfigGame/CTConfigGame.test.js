import React from 'react';
import { shallow } from 'enzyme';
import CTConfigGame from './CTConfigGame';

it('renders without crashing with all required props', () => {
  const component = shallow(
    <CTConfigGame lower={1} onReset={() => {}} upper={10} />
  );
  expect(component.debug()).toMatchSnapshot();
});

it('handles parsing value of changing lower number', () => {
  const component = shallow(<CTConfigGame lower={1} onReset={() => {}} upper={10} />);
  expect(component.state('userLower')).toEqual('');
  component.instance().onChangeLower({ target: { value: '5' }});
  expect(component.state('userUpper')).toEqual('');
  expect(component.state('userLower')).toEqual(5);
});

it('handles parsing value of changing upper number', () => {
  const component = shallow(<CTConfigGame lower={1} onReset={() => {}} upper={10} />);
  expect(component.state('userUpper')).toEqual('');
  component.instance().onChangeUpper({ target: { value: '5' }});
  expect(component.state('userUpper')).toEqual(5);
  expect(component.state('userLower')).toEqual('');
});

it('calls parent reset when onReset() is invoked', () => {
  const mockFunction = jest.fn();
  const component = shallow(<CTConfigGame lower={1} onReset={mockFunction} upper={10} />);
  component.instance().onReset();
  expect(mockFunction.mock.calls.length).toBe(1);
});
