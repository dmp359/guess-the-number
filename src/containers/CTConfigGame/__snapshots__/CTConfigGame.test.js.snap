exports[`test renders without crashing with all required props 1`] = `
"<div className=\"config\">
  <div className=\"config-label\">
    Game Config
  </div>
  <LabelBox label=\"Lower bound: \" onChange={[Function: bound onChangeLower]} value={1} />
  <LabelBox label=\"Upper bound: \" onChange={[Function: bound onChangeUpper]} value={10} />
  <button onClick={[Function: bound onReset]}>
    Reset
  </button>
</div>"
`;
